lib/auth.guard.ts

```diff
@@ -34,11 +34,13 @@ function createAuthGuard(type?: string | string[]): Type<CanActivate> {
     @Optional()
     @Inject(AuthModuleOptions)
     protected options: AuthModuleOptions = {};
+    
+    protected logger = new Logger('AuthGuard');
```

Comment:
I think I just recalled myself why `new Logger` was inlined before @jmcdo29 @TBG-FR. Now if any class extends the `AuthGuard` and already has a logger property that is private, TS compilation will error out with the following message:

> Class '...' incorrectly extends base class 'AuthGuard'.
>  Property 'logger' is private in type '...' but not in type 'AuthGuard'

consequently, this PR introduces a breaking change now

Reply:
Oh, I see, didn't think of that issue ! So should we switch back to inline instanciation, or do we have a better alternative ?

Reply:
We could just create a logger instance as a local variable above the `createAuthGuard` function declaration